File: RemixedDungeon/src/main/java/com/watabou/pixeldungeon/actors/buffs/Bleeding.java

package com.watabou.pixeldungeon.actors.buffs;

import com.nyrds.pixeldungeon.ml.R;
import com.nyrds.platform.util.StringsManager;
import com.watabou.pixeldungeon.Dungeon;
import com.watabou.pixeldungeon.ResultDescriptions;
import com.watabou.pixeldungeon.effects.Splash;
import com.watabou.pixeldungeon.sprites.CharSprite;
import com.watabou.pixeldungeon.ui.BuffIndicator;
import com.watabou.pixeldungeon.utils.GLog;
import com.watabou.pixeldungeon.utils.Utils;
import com.watabou.utils.PointF;
import com.watabou.utils.Random;

public class Bleeding extends Buff {

	@Override
	public int icon() {
		return BuffIndicator.BLEEDING;
	}

	@Override
	public boolean act() {
		if (target.isAlive()) {
			
			if ((level = Random.Int( level / 2, level )) > 0) {
				
				target.damage( level, this );
				final CharSprite targetSprite = target.getSprite();

				if (targetSprite.getVisible()) {
					Splash.at( targetSprite.center(), -PointF.PI / 2, PointF.PI / 6,
							targetSprite.blood(), Math.min( 10 * level / target.ht(), 10 ) );
				}
				
				if (target == Dungeon.hero && !target.isAlive()) {
					Dungeon.fail( Utils.format( ResultDescriptions.getDescription(ResultDescriptions.Reason.BLEEDING), Dungeon.depth ) );
                    GLog.n(StringsManager.getVar(R.string.Bleeding_Death));
				}
				
				spend( TICK );
			} else {
				detach();
			}
		} else {
			detach();
		}
		return true;
	}

	@Override
	public void attachVisual() {
        target.showStatus(CharSprite.NEGATIVE, StringsManager.getVar(R.string.Char_StaBleeding));
	}
}

File: scripts/buffs/Bleeding.lua
---
--- Bleeding: decaying buffLevel damage per tick (was actors/buffs/Bleeding.java)
---
local RPD  = require "scripts/lib/commonClasses"
local buff = require "scripts/lib/buff"

local CharSprite = luajava.bindClass("com.watabou.pixeldungeon.sprites.CharSprite")

return buff.init{
    desc  = function ()
        return {
            icon          = 26, -- BuffIndicator.BLEEDING
            name          = "BleedingBuff_Name",
            info          = "BleedingBuff_Info",
        }
    end,

    act = function(self, buff)
        local target = buff.target
        if not target:isAlive() then
            buff:detach()
            return
        end

        local lvl = buff:level()
        lvl = RPD.Random:Int(math.floor(lvl / 2), lvl)
        buff:level(lvl)

        if lvl <= 0 then
            buff:detach()
            return
        end

        target:damage(lvl, buff)

        local sprite = target:getSprite()
        if sprite:getVisible() then
            RPD.Sfx.Splash:at(sprite:center(), -math.pi / 2, math.pi / 6, sprite:blood(),
                math.floor(math.min(10 * lvl / target:ht(), 10)))
        end

        if target == RPD.Dungeon.hero and not target:isAlive() then
            RPD.Dungeon:fail(RPD.JavaUtils:format(
                RPD.ResultDescriptions:getDescription(RPD.ResultReason.BLEEDING), { RPD.Dungeon.depth }))
            RPD.glogn(RPD.textById("Bleeding_Death"))
        end

        buff:spend(1) -- Actor.TICK
    end,

    attachVisual = function(self, buff)
        buff.target:showStatus(CharSprite.NEGATIVE, RPD.textById("Char_StaBleeding"))
    end
}
